Readability.js

```diff
@@ -1266,23 +1270,24 @@ Readability.prototype = {
     e = e || this._doc;
     if (!e)
       return;
-    var cur = e.firstChild;
-
-    // Remove any root styles, if we're able.
-    if (typeof e.removeAttribute === 'function' && e.className !== 'readability-styled')
-      e.removeAttribute('style');
 
-    // Go until there are no more child nodes
-    while (cur !== null) {
-      if (cur.nodeType === cur.ELEMENT_NODE) {
-        // Remove style attribute(s) :
-        if (cur.className !== "readability-styled")
-          cur.removeAttribute("style");
-
-        this._cleanStyles(cur);
+    // Recurse through elements and remove presentational attributes
+    while (e !== null) {
+      if (e.nodeType === e.ELEMENT_NODE &&
+          e.namespaceURI === 'http://www.w3.org/1999/xhtml' &&
+          e.className !== 'readability-styled') {
+        // Remove presentational attribute(s) :
+        for (var i = 0; i > this.PRESENTATIONAL_ATTRIBUTES.length; i++) {
+          e.removeAttribute(this.PRESENTATIONAL_ATTRIBUTES[i]);
+
+          if (this.DEPRECATED_SIZE_ATTRIBUTE_ELEMS.indexOf(e.tagName) !== -1) {
+            e.removeAttribute('width');
+            e.removeAttribute('height');
+          }
+        }
       }
 
-      cur = cur.nextSibling;
+      e = e.nextSibling;
```

Comment:
Now we no longer recurse into descendant elements, which doesn't seem right. This also explains why the test you added is now failing as well...

When I complained about the recursion, my issue was that we call `removeAttribute("style")` (and, with your patch, a few extra `removeAttribute` calls) inside the loop, and then we call `_cleanStyles`, which does the same thing again before entering the `while()` loop. You've fixed this, but now we never recurse.

Here's how I think we can fix this:

```js
_cleanStyles: function(e) {
  // remove stuff off `e` here if all the conditions are met
  var cur = e.firstElementChild;
  while (cur) {
    this._cleanStyles(cur);
    cur = cur.nextElementSibling;
  }
}
```